File: 532_K-diff_Pairs_in_a_Array.java
// 注意考虑k = 0 的情况
public class Solution {
    public int findPairs(int[] nums, int k) {
        Map<Integer, Integer> map = new HashMap<>();
        for (int n: nums) {
            map.put(n, map.getOrDefault(n, 0) + 1);
        }
        int count = 0;
        for (int key : map.keySet()) {
            if (k == 0) {
                if (map.get(key) > 1) 
                    count++;
            } else if (k > 0) {
                if (map.containsKey(key + k))
                    count++;
            }
        }
        return count;
    }
}

File: 532_K-diff_Pairs_in_a_Array.py
class Solution(object):
    def findPairs(self, nums, k):
        """
        :type nums: List[int]
        :type k: int
        :rtype: int
        """
        if k == 0:
            return sum(v>1 for v in collections.Counter(nums).values())
        elif k > 0:
            return len(set(nums)&set(n+k for n in nums))
        else:
            return 0
        # nums_set = set(nums)
        # count = 0
        # for n in nums_set:
        #     if n + k in  nums_set:
        #         count += 1
        # return count
        